### PR TITLE
Disable linux-x64 dev-innerloop

### PR DESCRIPTION
**Pull request by am11**
Lately, this leg just times out with pending status on GItHub:

>##[error]
>,##[error]Agent failed with exception: The machine running request aa3690cc-533c-40f0-8670-ebfaae75e388 restarted. Azure DevOps can't recover from restarts.
,##[warning]Received request to deprovision: The request was cancelled by the remote provider.

From https://github.com/dotnet/dnceng/issues/3879, it is currently the only leg running into this issue.

### CODE REVIEW

**Comment by steveisok** [comment]
I'm fine with disabling until we either fix the mega-build into pieces or bump the memory.

**Comment by am11** [comment]
**Fix the mega-build by breaking it into smaller pieces.**

Cc: @ericstj. This is related to the OOM issue we encountered with the `-allConfigurations` leg running on Linux (see https://github.com/dotnet/dnceng/issues/3879#issuecomment-2396858080). 

Could we consider splitting the build into groups? For example, we could run multiple `dotnet` invocations and terminate the process after each one to manage memory usage more effectively.

**Comment by am11** [comment]
> Fix the mega-build by breaking it into smaller pieces.

Cc: @ericstj. This is related to the OOM issue we encountered with the `-allConfigurations` leg running on Linux (see https://github.com/dotnet/dnceng/issues/3879#issuecomment-2396858080). 

Could we consider splitting the build into groups? For example, we could run multiple `dotnet` invocations and terminate the process after each one to manage memory usage more effectively.

**Comment by jkotas** [comment]
> This is related to the OOM issue we encountered with the -allConfigurations leg running 

Do we know what consumes the memory and whether it is expected? We should check that this is not a symptom of a product issue.

**Comment by steveisok** [comment]
> > This is related to the OOM issue we encountered with the -allConfigurations leg running
> 
> Do we know what consumes the memory and whether it is expected? We should check that this is not a symptom of a product issue.

@agocke suggested we may have broken something in main and to see how it behaves after https://github.com/dotnet/runtime/pull/107772 went in.

**Comment by jkotas** [comment]
> behaves after https://github.com/dotnet/runtime/pull/107772 went in.

linux-x64 dev-innerloop has been failing for like a month. It started a long before #107772 .

**Comment by ericstj** [comment]
Agree with what @jkotas says.  There's no reason for us to think that the product shouldn't be capable of building the `-allconfigurations` leg.  If we're hitting OOMs that's a symptom of a memory leak in the build processes.

**Comment by am11** [comment]
> There's no reason for us to think that the product shouldn't be capable of building the `-allconfigurations` leg

CI machine is running out of memory after building assemblies for 50-60 mins. Locally builds succeeds because I have 32 gigs on my linux box (and not many running processes). I don't think anyone was suggesting that the entire product is incapable..

**Comment by jkotas** [comment]
> CI machine is running out of memory after building assemblies for 50-60 mins

Right, it suggests that there is "leak" that accumulates to be big-enough to take the CI machine down with long-enough build. A single assembly takes no more than 10's seconds to build. All memory required to build given assembly should be released after we are done building it if there is a memory pressure.

**Comment by ericstj** [comment]
I pulled down the log files here and the thing that looks suspect here is some hotreload tasks.  I made a mention in the issue.  
image

Totally non-scientific but my sampling of logs from passing builds always showed these tasks running around the time we see 95% memory warnings.  Could be relevant.

This build leg isn't just building allConfigurations, it's also building all the tests.  Not sure if that's relevant but it is a bit unusual.  Probably gives the system the biggest load of concurrent builds of managed code we have to offer.

> linux-x64 dev-innerloop has been failing for like a month.

Could it be correlated to us updating the SDK to RC1?

**Comment by am11** [comment]
@jkotas should we merge this until https://github.com/dotnet/dnceng/issues/3879 is resolved since it's failure rate is high these days? If you think we should keep it enabled, lets close this PR and track it in the issue.

cc @dotnet/dotnet-hotreload-utils-admin

**Comment by jkotas** [comment]
I think we should have this leg disabled against an issue in this repo that is specific to linux_x64_dev_innerloop

https://github.com/dotnet/dnceng/issues/3879 looks too generic, not very actionable.

File: eng/pipelines/common/platform-matrix.yml [hunk 1]
```diff
@@ -298,23 +298,23 @@ jobs:
...
-      targetRid: linux-x64
-      platform: linux_x64
-      shouldContinueOnError: ${{ parameters.shouldContinueOnError }}
-      container: linux_x64_dev_innerloop
-      jobParameters:
-        runtimeFlavor: ${{ parameters.runtimeFlavor }}
-        buildConfig: ${{ parameters.buildConfig }}
-        helixQueueGroup: ${{ parameters.helixQueueGroup }}
-        ${{ insert }}: ${{ parameters.jobParameters }}
+#- ${{ if containsValue(parameters.platforms, 'linux_x64_dev_innerloop') }}:
```
Comment (am11) [inline eng/pipelines/common/platform-matrix.yml:301]:
```suggestion
# disabled against https://github.com/dotnet/runtime/issues/108821
#- ${{ if containsValue(parameters.platforms, 'linux_x64_dev_innerloop') }}:
```

**Comment by am11** [comment]
Opened https://github.com/dotnet/runtime/issues/108821.

**Review by jkotas — APPROVED** [review]

**Comment by steveisok** [comment]
Next steps are figuring out how we're going to go about investigating and who is going to do it.

**Pull request merged**